src/main/java/com/google/jenkins/plugins/storage/DownloadStep.java

```diff
@@ -306,19 +309,38 @@ public void checkRoles(RoleChecker checker)
   }
 
   /**
+   * A class to store StorageObject information in a serializable manner.
+   *
+   */
+  protected static class StorageObjectId implements Serializable {
+    public StorageObjectId(StorageObject obj) {
+      this.bucket = obj.getBucket();
+      this.name = obj.getName();
+    }
+
+    public String getBucket() {
+      return bucket;
+    }
+
+    public String getName() {
+      return name;
+    }
+
+    private final String bucket;
+    private final String name;
+  }
+
+  /**
    * DownloadSpec is a way to encode what needs to be downloaded.
    */
   protected static class DownloadSpec implements Serializable {
```

Comment:
Is this class needed anymore? It looks like you could just pass around the list of StorageObjectIds.

Reply:
Good point